core/src/main/scala/akka/persistence/dynamodb/internal/JournalDao.scala

```diff
@@ -134,83 +137,113 @@ import software.amazon.awssdk.services.dynamodb.model.Update
       attributes
     }
 
-    val totalEvents = events.size
-    if (totalEvents == 1) {
-      val req = PutItemRequest
-        .builder()
-        .tableName(settings.journalTable)
-        .item(putItemAttributes(events.head))
-        .returnConsumedCapacity(ReturnConsumedCapacity.TOTAL)
-        .build()
-      val result = client.putItem(req).asScala
+    val totalSize = events.iterator.map(_.payload.fold(0)(_.size)).sum
 
-      if (log.isDebugEnabled()) {
-        result.foreach { response =>
-          log.debug(
-            "Wrote [{}] events for persistenceId [{}], consumed [{}] WCU",
-            1,
-            persistenceId,
-            response.consumedCapacity.capacityUnits)
-        }
+    if (s3Fallback.nonEmpty &&
+      totalSize > settings.s3FallbackSettings.threshold &&
+      events.forall(_.serManifest != S3FallbackSerializer.BreadcrumbManifest)) {
```

Comment:
I think it would be better to look at the serializer id (or id and manifest). That way you can also make the manifest a short string.